fe_utils/finite_elements.py

```diff
@@ -37,7 +50,40 @@ def vandermonde_matrix(cell, degree, points, grad=False):
     <ex-vandermonde>`.
     """
 
-    raise NotImplementedError
+    #1-dimensional cell
+    if cell.dim == 1: 
+        #Initialization
+        points = points.reshape(-1,)
+        m, = np.shape(points)
+        van = np.empty(shape=[m, 0])
+
+        for i in range(degree+1):
+            #construct whole column
+            col = np.array([x**i for x in points])
+            col = col.reshape(-1,1)
+            #stack the column
+            van = np.hstack([van, col])
+                
+        return van
+          
+    #2-dimensional cell
+    elif cell.dim == 2: 
+        #Initialization
+        m = np.shape(points)[0]
+        van = np.empty(shape=[m, 0])
+
+        for i in range(degree+1):
+            for j in range(i+1):
+                #construct whole column
+                col = np.array([coord[0]**(i-j) * coord[1]**j for coord in points])
```

Comment:
You can actually do these steps for all points at once. There is no need to explicitly loop over the points.